toolchain/lower/file_context.cpp

```diff
@@ -146,6 +160,240 @@ auto FileContext::Run() -> std::unique_ptr<llvm::Module> {
   return std::move(llvm_module_);
 }
 
+auto FileContext::PairInserted(
+    SemIR::SpecificId specific_id1, SemIR::SpecificId specific_id2,
+    Set<std::pair<SemIR::SpecificId, SemIR::SpecificId>>& set_of_pairs)
+    -> bool {
+  if (specific_id1.index > specific_id2.index) {
+    std::swap(specific_id1.index, specific_id2.index);
+  }
+  std::pair<SemIR::SpecificId, SemIR::SpecificId> pair = {specific_id1,
+                                                          specific_id2};
+  auto insert_result = set_of_pairs.Insert(pair);
```

Comment:
Not sure if you'd considered this, but maybe:

```suggestion
  auto insert_result = set_of_pairs.Insert({specific_id1, specific_id2});
```

Note, you could also consider taking id1 and id2 as a `std::pair` parameter instead of separately.

Reply:
Type cannot be inferred here, switching to make_pair.